mne/preprocessing/artifact_detection.py

```diff
@@ -0,0 +1,214 @@
+# Authors: Adonay Nunes <adonay.s.nunes@gmail.com>
+#          Luke Bloy <luke.bloy@gmail.com>
+# License: BSD (3-clause)
+
+
+from scipy.ndimage.measurements import label
+import numpy as np
+from ..annotations import (Annotations, _annotations_starts_stops)
+from ..chpi import _apply_quat
+from ..transforms import (quat_to_rot, _average_quats)
+from .. import Transform
+from ..utils import _mask_to_onsets_offsets
+
+
+def annotate_movement(raw, pos, rotation_velocity_limit=None,
+                      translation_velocity_limit=None,
+                      mean_distance_limit=None):
+    """Detect segments with movement.
+
+    Detects segments periods further from `rotation_velocity_limit`,
+    `translation_velocity_limit` and `mean_distance_limit`. It returns an
+    annotation with the bad segments.
+
+    Parameters
+    ----------
+    raw instance : instance of Raw. Data to detect head movement for.
+        Instance of a raw.
+    pos : array, shape (N, 10)
+        The position and quaternion parameters from cHPI fitting.
+    rotation_velocity_limit: float
+        Head rotation velocity limit in radians per second.
+    translation_velocity_limit : float
+        Head translation velocity limit in radians per second.
+    mean_distance_limit : float
+        Head position limit from mean recording in meters.
+
+    Returns
+    -------
+    annot : mne.Annotations
+        Periods with head motion.
+    hpi_disp : array
+        Head position over time with respect to the mean head pos.
+    """
+    sfreq = raw.info['sfreq']
+    hp_ts = pos[:, 0].copy()
+    hp_ts -= raw.first_samp / sfreq
+    dt = np.diff(hp_ts)
+    seg_good = np.append(dt, 1. / sfreq)
+    hp_ts = np.concatenate([hp_ts, [hp_ts[-1] + 1. / sfreq]])
+
+    annot = Annotations([], [], [], orig_time=None)  # rel to data start
+
+    # Mark down times that are bad according to annotations
+    onsets, ends = _annotations_starts_stops(raw, 'bad')
+    for onset, end in zip(onsets, ends):
+        seg_good[onset:end] = 0
+
+    # Annotate based on rotational velocity
+    t_tot = raw.times[-1]
+    if rotation_velocity_limit is not None:
+        from mne.transforms import _angle_between_quats
+        assert rotation_velocity_limit > 0
+        # Rotational velocity (radians / sec)
+        r = _angle_between_quats(pos[:-1, 1:4], pos[1:, 1:4])
+        r /= dt
+        bad_mask = (r >= np.deg2rad(rotation_velocity_limit))
+        onsets, offsets = _mask_to_onsets_offsets(bad_mask)
+        onsets, offsets = hp_ts[onsets], hp_ts[offsets]
+        bad_pct = 100 * (offsets - onsets).sum() / t_tot
+        print(u'Omitting %5.1f%% (%3d segments): '
```

Comment:
use `logger.info()` instead of printing - this way it can be controlled by verbose option, see for example how it's done in filtering: https://github.com/mne-tools/mne-python/blob/master/mne/filter.py#L662